python-package/mlbox/encoding/categorical_encoder.py

```diff
@@ -94,30 +94,29 @@ def fit(self, df_train, y_train):
         self.__Lcat = df_train.dtypes[df_train.dtypes == 'object'].index
         self.__Lnum = df_train.dtypes[df_train.dtypes != 'object'].index
 
+        if (len(self.__Lcat) == 0):
+            pass
+
```

Comment:
@AxeldeRomblay 

I've taken out the ` if (len(self.__Lcat) == 0)` condition from all the cases below because we only need to check it once at the beginning of the method.

If the condition holds true, then `pass` directly instead of looking at the strategy chosen.